packages/app/lib/ui/widgets/location_field.dart

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:app/ui/widgets/card.dart';
+import 'package:app/ui/widgets/card_action_button.dart';
+import 'package:app/ui/widgets/card_header.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:maplibre_gl/maplibre_gl.dart';
+
+typedef OnUpdate = void Function(
+    {required double latitude, required double longitude});
+
+class LocationField extends StatefulWidget {
+  final double latitude;
+  final double longitude;
+
+  final OnUpdate onUpdate;
+
+  const LocationField(
+      {super.key,
+      required this.longitude,
+      required this.latitude,
+      required this.onUpdate});
+
+  @override
+  State<LocationField> createState() => _LocationFieldState();
+}
+
+enum InputMode { read, edit }
+
+double DEFAULT_ZOOM = 10;
+String CURRENT_FEATURE_ID = 'current';
+String TARGET_FEATURE_ID = 'target';
+
+class _LocationFieldState extends State<LocationField> {
+  InputMode _inputMode = InputMode.read;
+
+  late double _lat;
+  late double _lon;
+
+  MaplibreMapController? _mapController;
+  Circle? _currentMapMarker;
+
+  @override
+  void initState() {
```

Comment:
When no lat or lon has been set on creating a sighting, it would be good if the default position was somewhere in Brazil :smile: and then maybe the default zoom also much further out.